service/controller/clusterapi/v29/machine_deployment_resource_set.go

```diff
@@ -31,6 +30,22 @@ func NewMachineDeploymentResourceSet(config MachineDeploymentResourceSetConfig)
 		}
 	}
 
+	var awsClientResource controller.Resource
+	{
+		c := awsclient.Config{
+			K8sClient:     config.K8sClient,
+			Logger:        config.Logger,
+			ToClusterFunc: newMachineDeploymentToClusterFunc(config.CMAClient),
```

Comment:
The transform function is either the usual key function or some lookup function we use already.